tuf/api/metadata.py

```diff
@@ -720,15 +728,13 @@ def from_dict(cls, signed_dict: Dict[str, Any]) -> "Timestamp":
     def to_dict(self) -> Dict[str, Any]:
         """Returns the dict representation of self."""
         res_dict = self._common_fields_to_dict()
-        res_dict["meta"] = {
-            "snapshot.json": self.meta["snapshot.json"].to_dict()
-        }
+        res_dict["meta"] = {"snapshot.json": self.snapshot_meta.to_dict()}
         return res_dict
 
     # Modification.
     def update(self, snapshot_meta: MetaFile) -> None:
         """Assigns passed info about snapshot metadata to meta dict."""
-        self.meta["snapshot.json"] = snapshot_meta
+        self.snapshot_meta = snapshot_meta
```

Comment:
I think this pretty clearly shows that the update() function has no utility: your property setter now does the actual work with better ergonomics.

I think we should remove update()

Reply:
Will do that change in a separate pr where we can change all of the update functions
and propose a better solution.